python/ray/serve/_private/replica_scheduler/replica_scheduler.py

```diff
@@ -677,8 +836,14 @@ def update_running_replicas(self, running_replicas: List[RunningReplicaInfo]):
             [self.create_replica_wrapper(r) for r in running_replicas]
         )
 
-    async def choose_replicas_with_backoff(
+    @abstractmethod
+    async def choose_replicas(
         self,
-        request_metadata: Optional[RequestMetadata] = None,
-    ) -> AsyncGenerator[List[RunningReplicaInfo], None]:
+        replicas_ranks: List[List[RunningReplica]],
+        pending_request: Optional[PendingRequest] = None,
+    ) -> List[List[RunningReplica]]:
+        """One iteration of choosing available replicas.
+        This is the main function each custom scheduler should implement to
+        decide which replica to send the request.
+        """
```

Comment:
And please include descriptions for args `replicas_ranks` and `pending_request` in docstring
```suggestion
        """Chooses a subset of candidate replicas from available replicas.

        This is the main function each replica scheduler should implement to
        decide which replica to send the request to. This is one iteration of
        replica selection.
        """
```